internal/controllers/nodefeaturediscovery_reconciler.go

```diff
@@ -42,8 +43,8 @@ type nodeFeatureDiscoveryReconciler struct {
 	helper nodeFeatureDiscoveryHelperAPI
 }
 
-func NewNodeFeatureDiscoveryReconciler(client client.Client, deploymentAPI deployment.DeploymentAPI, scheme *runtime.Scheme) nodeFeatureDiscoveryReconciler {
-	helper := newNodeFeatureDiscoveryHelperAPI(client, deploymentAPI, scheme)
+func NewNodeFeatureDiscoveryReconciler(client client.Client, deploymentAPI deployment.DeploymentAPI, workerAPI worker.WorkerAPI, scheme *runtime.Scheme) nodeFeatureDiscoveryReconciler {
```

Comment:
IMHO it is better to use daemonset/deployment/configmap packages than worker/topology/etc'. Please see my [PR](https://github.com/kubernetes-sigs/node-feature-discovery-operator/pull/217).  We daemonsets/configmaps/deployments are the building block: we have daemonsets both for topology and workers, so i think it makes sense to set all daemonset under specific package etc'. Also, in case any other  future component will need to be deployed by the operator, we can just use the building blocks as is, with minimal changes. WDYT?

Reply:
Sounds sensible to me (if re-usable building blocks like "daemonset" can be easily made)